app/services/idv/profile_step.rb

```diff
@@ -1,61 +1,71 @@
 module Idv
   class ProfileStep < Step
-    def complete?
-      idv_session.resolution.try(:success?) ? true : false
-    end
+    def submit
+      initialize_idv_session
 
-    def complete
-      return track_event if attempts_exceeded?
-      idv_session.params.merge!(params)
-      vendor_validate if form_validate(params)[:success]
-      track_event
-      complete?
+      @success = complete?
+
+      increment_attempts_count if form_valid?
+      update_idv_session
+
+      FormResponse.new(success: success, errors: errors, extra: extra_analytics_attributes)
     end
 
     def attempts_exceeded?
       attempter.exceeded?
     end
 
+    def duplicate_ssn?
+      errors.key?(:ssn) && errors[:ssn].include?(I18n.t('idv.errors.duplicate_ssn'))
+    end
+
     private
 
+    attr_reader :success
+
+    def increment_attempts_count
+      attempter.increment
+    end
+
     def attempter
       @_idv_attempter ||= Idv::Attempter.new(idv_form.user)
     end
 
+    def initialize_idv_session
+      idv_session.params.merge!(params)
+      idv_session.applicant = vendor_params
+    end
+
     def vendor_params
       idv_session.applicant_from_params
     end
 
-    def vendor_validate
-      result = vendor_validator.validate
-      attempter.increment
-      result
+    def complete?
```

Comment:
can we put this above `increment_attempts_count` since it is used first? I usually put private methods in order of usage so it's easier to find them.